torch/distributed/_tensor/ops/tensor_ops.py

```diff
@@ -290,60 +210,47 @@ def prop_slice_scatter(op_schema: OpSchema) -> OutputSharding:
     #   where our best bet for now is to make them replicated as a fall-back.
     #   TODO: Ideally we'd like to make sure the output is re-sharded afterwards to keep input sharding.
 
-    defaults = (None, None, 0, None, None, 1)
-    input, src, dim, start, end, step = (
-        op_schema.args_schema + defaults[len(op_schema.args_schema) :]
-    )
-    assert isinstance(input, DTensorSpec)
-    assert isinstance(src, DTensorSpec)
-    assert isinstance(dim, int)
-
-    if dim < 0:
-        dim += input.ndim
+    input_node = cast(Node, node.args[0])
+    input_ndim = input_node.meta["val"].ndim
+    slice_dim = cast(int, node.args[2]) if len(node.args) > 2 else 0
+    slice_dim = normalize_dim(slice_dim, input_ndim)
+    input_strategy = node_to_strategy[input_node]
+    assert isinstance(input_strategy, OpStrategy)
 
-    # first, we keep the input sharding, except for the input dimension
-    # also, we cannot allow partial sum anymore.
-    input_suggestion = tuple(
-        Replicate()
-        if isinstance(p, _Partial) or (isinstance(p, Shard) and p.dim == dim)
-        else p
-        for p in input.placements
-    )
+    slice_scatter_strategy = OpStrategy([])
+    # by default follow the input strategy for both input and src
+    for arg_strategy in input_strategy.strategies:
+        arg_spec = arg_strategy.output_spec
+        if not (
+            is_tensor_dim_sharded(arg_spec, dim=slice_dim)
+            or is_tensor_partial(arg_spec)
+        ):
+            # only add the strategy if the slice_scatter dim is not sharded or partial
+            slice_scatter_strategy.strategies.append(
+                PlacementStrategy(output_spec=arg_spec)
+            )
 
-    if input_suggestion == tuple(input.placements) and src.placements == tuple(
-        input.placements
-    ):
-        # if our sharding is correct, the output sharding will be the same as the input.
-        return OutputSharding(
-            output_spec=DTensorSpec(
-                mesh=input.mesh,
-                placements=input.placements,
+    if not slice_scatter_strategy.strategies:
+        # if all strategies are filtered out, replicating all specs on slice_scatter dim
+        # of the input strategy, and use that as the op strategy
+        for arg_strategy in input_strategy.strategies:
+            arg_spec = arg_strategy.output_spec
+            replicate_spec = DTensorSpec(
+                mesh, replicate_tensor_dim(arg_spec.placements, dim=slice_dim)
```

Comment:
the difference between `replicate_tensor_dim` and `unshard_tensor_dim` is, `unshard_tensor_dim` only replicate when the dim is `Shard` while `replicate_tensor_dim` replicate when the dim is either `Shard` or `_Partial`. What is the reason doing this?

Reply:
`replicate_tensor_dim` means if tensor dim is sharded we replicate it, also not allowing partial, `unshard` tensor dim only checks if it's sharded but not check partial, this is to deal with cases are not accepting partial, i.e. `slice_scatter` not accepting partial